libp2p/Host.cpp

```diff
@@ -109,6 +110,10 @@ Host::~Host()
 void Host::start()
 {
     DEV_TIMED_FUNCTION_ABOVE(500);
+    auto nodeTable = this->nodeTable();
+    if (nodeTable)
+        BOOST_THROW_EXCEPTION(NetworkRestartNotSupported());
```

Comment:
Can you explain this part? This requires a mutex lock and a shared_ptr copy.

Reply:
Ah I don’t suppose I need to take the lock since I’m just checking if the handle is valid and don’t care about the state of the buckets or node list. I’ll change this before merging.